packages/core/http/core-http-server-internal/src/http_server.ts

```diff
@@ -185,7 +186,8 @@ export class HttpServer {
   constructor(
     private readonly logger: LoggerFactory,
     private readonly name: string,
-    private readonly shutdownTimeout$: Observable<Duration>
+    private readonly shutdownTimeout$: Observable<Duration>,
+    private readonly env: Env
```

Comment:
NIT: I would have either passed the whole `core: CoreContext`, or the fully specialized `buildSha` (given it's the only thing we're using). Passing a middle layer structure like the `Env` feels slightly weird to me.

But it's probably still fine